### PR TITLE
Implement basic launching functionality for desktop

### PR DESCRIPTION
**Pull request by CarbonSmasher**
I have brought the desktop client to a state where it is able to launch Smithed bundles.

Features:
- Game launching with the latest version of MCVM
- Local pack bundles that can be launched with the contained packs automatically installed. Data is kept separate from other bundles.
- Browse features, such as adding packs on pack pages to local bundles, or importing bundles from user pages locally.
- Global loading of data and resource packs using the Fabric loader and Paxi.
- Creation, pack addition and deletion, and removal of local bundles.

What still needs to be done:
- Paxi only supports 1.19.2+, which is not nearly all of the Smithed supported versions. Once companion is ready, we can use that. Radon would also be nice to add for optimization.
- I added some Font Awesome icons that should probably be replaced with something better.
- I tried my best to make things look decent, but some UI design work is still needed.
- Keeping the user logged in to their Microsoft account while the app is open.

This PR isn't perfect, but I hope it's helpful in moving Smithed along as a platform. I got some great experience with React and Tauri and got to see firsthand how MCVM worked as a library :)

### CODE REVIEW

File: packages/components/assets/save.svg [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?><!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
-<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" fill="none">
+<!-- <?xml version="1.0" encoding="utf-8"?>Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
+<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" style="filter: invert(100%) sepia(100%) saturate(0%) hue-rotate(174deg) brightness(102%) contrast(102%);">
```
Comment (TheNuclearNexus) [inline packages/components/assets/save.svg:2]: Why is a filter being applied here? Applying these in the svg file causes them to change everywhere

File: packages/components/BackButton.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { IconTextButton } from "./IconTextButton";
+import { Left, Right } from "./svg";
+import "./BackButton.css";
+
+function BackButton() {
+	return (
+		<div className="backButton">
+			<IconTextButton
+				text="Back"
+				href="javascript:window.history.back();"
```
Comment (TheNuclearNexus) [inline packages/components/BackButton.tsx:10]: You should be using the React Router `useHistory()` hook, weird javascript isn't great

File: packages/components/ChooseBox.tsx [hunk 3]
```diff
@@ -76,26 +80,29 @@ export function ChooseBox({ placeholder, defaultValue, multiselect, choices, onC
...
-                {choices.map(c => <div key={c.value} onMouseDown={(e) => clickOption(e, c.value)} className="chooseBoxOption">
-                    {c.content}
-                    {(multiselect ? value.includes(c.value) : value === c.value) && (multiselect ?<Check/> : <svg xmlns="http://www.w3.org/2000/svg" width="9" height="8" viewBox="0 0 9 8" fill="none">
-                        <circle cx="4.5" cy="4" r="4" fill="#FFF8F0" />
-                    </svg>)}
-                </div>)}
-            </div>
+            <Right style={{ transform: `rotate(${open ? (flip ? '-90' : '90') : (flip ? '90' : '-90')}deg)`, transition: 'all 0.25s ease-in-out', flexShrink: 0 }} />
         </div>
+        {!flip && options}
```
Comment (TheNuclearNexus) [inline packages/components/ChooseBox.tsx:106]: This is fine but generally you should just use `flex-direction: reverse-column`

File: packages/components/IconTextButton.tsx [hunk 4]
```diff
@@ -1,18 +1,54 @@
...
+			className={
+				`buttonLike${props.disabled ? " disabled" : ""} ` + props.className
+			}
+			style={{ flexDirection: reverse ? "row-reverse" : "row", ...style }}
+			href={href}
+			target={target}
+			onClick={props.onClick}
+			rel={props.rel}
+			onMouseEnter={props.onMouseEnter}
+			onMouseLeave={props.onMouseLeave}
```
Comment (TheNuclearNexus) [inline packages/components/IconTextButton.tsx:36]: props should be deconstructed for consistency with the other props

File: platforms/desktop/src/types.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	{ content: "1.17", value: "1.17" },
+	{ content: "1.17.1", value: "1.17.1" },
+	{ content: "1.18", value: "1.18" },
+	{ content: "1.18.1", value: "1.18.1" },
+	{ content: "1.18.2", value: "1.18.2" },
+	{ content: "1.19", value: "1.19" },
+	{ content: "1.19.4", value: "1.19.4" },
+	{ content: "1.20", value: "1.20" },
+	{ content: "1.20.1", value: "1.20.1" },
+];
```
Comment (TheNuclearNexus) [inline platforms/desktop/src/types.ts:24]: This needs to be generated from the data-types package

File: platforms/desktop/src/pages/launch/LaunchPage.css [hunk 6]
```diff
@@ -0,0 +1,87 @@
+.launchContentContainer {
```
Comment (TheNuclearNexus) [inline platforms/desktop/src/pages/launch/LaunchPage.css:1]: Using percentages for padding as well as using scale over font-size might cause weird behavior, consider using fixed rem values instead

**Review by TheNuclearNexus — CHANGES_REQUESTED** [review]

File: platforms/desktop/src/types.ts (hunk 5 above)
Comment (CarbonSmasher) [reply platforms/desktop/src/types.ts]: Replaced the constant with an Array.map from the supportedMinecraftVersions

File: platforms/desktop/src/pages/launch/LaunchPage.css (hunk 6 above)
Comment (CarbonSmasher) [reply platforms/desktop/src/pages/launch/LaunchPage.css:1]: Replaced with fixed padding and font sizes

File: packages/components/IconTextButton.tsx (hunk 4 above)
Comment (CarbonSmasher) [reply packages/components/IconTextButton.tsx]: Deconstructed all the props

File: packages/components/ChooseBox.tsx (hunk 3 above)
Comment (CarbonSmasher) [reply packages/components/ChooseBox.tsx]: Used a conditional style instead

File: packages/components/BackButton.tsx (hunk 2 above)
Comment (CarbonSmasher) [reply packages/components/BackButton.tsx]: Moved into widgets and replaced with useNavigate

File: packages/components/assets/save.svg (hunk 1 above)
Comment (CarbonSmasher) [reply packages/components/assets/save.svg]: Replaced these filters on the component files themselves with inline fills where they are used

**Pull request merged**